src/lib/functions/registry.ts

```diff
@@ -290,15 +290,11 @@ export class FunctionsRegistry {
    * function with the given name exists, returns an object with the function
    * and the route set to `null`. Otherwise, `undefined` is returned,
    *
-   * @param {string} url
+   * @param {string} urlPath
    * @param {string} method
    */
-  // @ts-expect-error TS(7006) FIXME: Parameter 'url' implicitly has an 'any' type.
-  async getFunctionForURLPath(url, method) {
-    // We're constructing a URL object just so that we can extract the path from
-    // the incoming URL. It doesn't really matter that we don't have the actual
-    // local URL with the correct port.
-    const urlPath = new URL(url, 'http://localhost').pathname
```

Comment:
![Uploading Screenshot 2023-11-27 at 16.23.40.png…]()

Reply:
There is nothing ensuring that `urlPath` is indeed a URL path and not a full URL, right? I have a feeling that this is precisely why we had this logic here.

Reply:
I've played around with it a bit more, and it seems like we had this in place to remove query parameters from the path. https://github.com/netlify/cli/pull/6212/commits/146cb80d7e270c44918dcc77f48e564c8f5b40e6 implements a fix that still does that